articles/ros_node_idl.md

```diff
@@ -97,59 +104,128 @@ This proposal has a number of potential upsides, but it also has some downsides
-However, the keyword is likely solidly descriptive enough to either be accepted as falling under [REP 149][rep149_export] or else to motivate an amendment of the REP.
+In the case of setup.py, the above must be done manually alongside the placement of the package's marker in the ament index.
 
-### IDL Schema
+### NoDL Schema
+
+An `.xsd` xml schema is provided alongside the NoDL implementation, version 1 of which is displayed below
+
+```xml
+<?xml version="1.1" encoding="UTF-8" standalone="no"?>
+<xs:schema xmlns:xs="http://www.w3.org/2001/XMLSchema">
+
+    <xs:element name="action">
+        <xs:complexType>
+            <xs:attribute name="name" type="xs:string" use="required" />
+            <xs:attribute name="type" type="xs:string" use="required" />
+            <xs:attribute name="server" type="xs:boolean" default="false" />
+            <xs:attribute name="client" type="xs:boolean" default="false" />
+        </xs:complexType>
+    </xs:element>
+
+    <xs:element name="parameter">
+        <xs:complexType>
+            <xs:attribute name="name" type="xs:string" use="required" />
+            <xs:attribute name="type" type="xs:string" use="required" />
+        </xs:complexType>
+    </xs:element>
+
+    <xs:element name="service">
+        <xs:complexType>
+            <xs:attribute name="name" type="xs:string" use="required" />
+            <xs:attribute name="type" type="xs:string" use="required" />
+            <xs:attribute name="server" type="xs:boolean" default="false" />
+            <xs:attribute name="client" type="xs:boolean" default="false" />
+        </xs:complexType>
+    </xs:element>
+
+    <xs:element name="topic">
+        <xs:complexType>
+            <xs:attribute name="name" type="xs:string" use="required" />
+            <xs:attribute name="type" type="xs:string" use="required" />
+            <xs:attribute name="publisher" type="xs:boolean" default="false" />
+            <xs:attribute name="subscription" type="xs:boolean" default="false" />
+        </xs:complexType>
+    </xs:element>
+
+    <xs:element name="node">
+        <xs:complexType>
+            <xs:choice minOccurs="1" maxOccurs="unbounded">
+                <xs:element ref="action" />
+                <xs:element ref="parameter" />
+                <xs:element ref="topic" />
+                <xs:element ref="service" />
+            </xs:choice>
+            <xs:attribute name="name" type="xs:string" use="required" />
+            <xs:attribute name="executable" type="xs:string" use="required" />
+        </xs:complexType>
+    </xs:element>
+
+    <xs:element name="interface">
+        <xs:complexType>
+            <xs:choice maxOccurs="unbounded">
+                <xs:element ref="node"></xs:element>
+            </xs:choice>
+            <xs:attribute name="version" fixed="1" use="required" />
+        </xs:complexType>
+    </xs:element>
+</xs:schema>
+```
```

Comment:
This would be easier to read using `include_relative`, is there a reason we can't use that, here? I'm also not completely convinced we want to maintain this here. Do other design docs follow the pattern of duplicating XSDs in here? We didn't do that for SROS2, for example. We shouldn't need to maintain the same file in multiple places.